src/drone.rs

```diff
@@ -32,16 +33,18 @@ impl DroneTrait for MyDrone {
                     // each match branch may call a function to handle it to make it more readable
 
                         //temporary and just for testing
-                        println!("received packet at drone {}", self.drone_id);
+                        println!("packet received from {}, Packet: {:?}, {:?}", self.drone_id, packet.session_id, packet.pack_type);
 
                         //remember to remove the underscores when you actually start using the variable ig
                         match &packet.pack_type {
                             PacketType::Nack(_nack)=>self.forward_packet(packet),
                             PacketType::Ack(_ack)=>self.forward_packet(packet),
-                            PacketType::MsgFragment(_fragment)=>self.forward_packet(packet),
+                            PacketType::MsgFragment(fragment)=>self.handle_MsgFragment(fragment.fragment_index, packet),
```

Comment:
Penso sia ridondante passare l'indice del frammento. Penso sarebbe più chiaro passare solo il packet

```suggestion
                            PacketType::MsgFragment(fragment)=>self.handle_MsgFragment(packet),
```

Reply:
ok, però in generale cercherei di evitare ottimizzazioni premature, specie se così piccole

Reply:
Potrebbe essere ma questo tecnicamente risparmia un march che poi devi fare dentro il codice della funzione..
Voglio dire: se tu passassi solo il packet nella funzione questa non ha idea del pattern march appena fatto sul tipo, quindi ci si ritrova a "ricontrollare" che quel packet sia effettivamente del tipo fragment. Tutto per ottenere il suo id.
In questo modo invece sfrutto il match appena fatto per copiare un u64 in modo da averlo già pronto all'uso, evitandomi di rifare il match e semplificando il codice

`
if let PacketType::MsgFragment(fragment) = packet.pack_type {
                let nack = Nack{
                    fragment_index: fragment.fragment_index,
                    nack_type: NackType::Dropped
                };
        
                packet.pack_type = PacketType::Nack(nack);
            }
`

che diventa 

`
            let nack = Nack{
                fragment_index: index,
                nack_type: NackType::Dropped
            };
    
            packet.pack_type = PacketType::Nack(nack);
`